playground.sh

```diff
@@ -18,186 +18,398 @@
+    if [ -n "${dockerComposeCommand}" ]; then
+      if ${dockerComposeCommand} ls | grep -q "${playgroundRuntimeName}"; then
+        echo "[INFO] gravitino-playground is running in Docker."
+        runtime="docker"
+        return
+      fi
+    fi
+  fi
+
+  # Check if gravitino-playground is running in K8s
+  if command -v kubectl >/dev/null 2>&1; then
+    if kubectl get namespace "${playgroundRuntimeName}" >/dev/null 2>&1; then
+      if kubectl -n "${playgroundRuntimeName}" get pods | grep -q "Running"; then
+        echo "[INFO] gravitino-playground is running in Kubernetes."
+        runtime="k8s"
+        return
+      fi
+    fi
+  fi
 }
 
 start() {
-  echo "INFO: Starting the playground..."
+  echo "[INFO] Starting the playground..."
+  echo "[INFO] The playground requires ${requiredCpuCores} CPU cores, ${requiredRamGB} GB of RAM, and ${requiredDiskSpaceGB} GB of disk storage to operate efficiently."
 
-  case "$runtime" in
+  checkCurrentRuntime
+  if [ -n "${runtime}" ]; then
+    echo "[ERROR] Playground is already running in ${runtime}. Please stop it first."
+    exit 1
+  fi
+
+  checkRuntime
+
+  case "${runtime}" in
   k8s)
     testK8s
     checkHelm
+    checkK8sDisk
+    checkK8sRam
+    checkK8sCpu
     ;;
   docker)
+    checkPortsInUse
     testDocker
-    checkCompose
-    ports=(8090 9001 3307 19000 19083 60070 13306 15342 18080 18888 19090 13000)
-    for port in "${ports[@]}"; do
-      checkPortInUse ${port}
-    done
+    checkDockerCompose
+    checkDockerDisk
+    checkDockerRam
+    checkDockerCpu
     ;;
   esac
 
-  cd ${playground_dir}
-  echo "Preparing packages..."
+  cd ${playground_dir} || exit 1
+  echo "[INFO] Preparing packages..."
   ./init/spark/spark-dependency.sh
   ./init/gravitino/gravitino-dependency.sh
   ./init/jupyter/jupyter-dependency.sh
 
-  case "$runtime" in
+  case "${runtime}" in
   k8s)
-    helm upgrade --install gravitino-playground ./helm-chart/ \
-      --create-namespace --namespace gravitino-playground \
+    helm upgrade --install ${playgroundRuntimeName} ./helm-chart/ \
+      --create-namespace --namespace ${playgroundRuntimeName} \
       --set projectRoot=$(pwd)
     ;;
   docker)
-    logSuffix=$(date +%Y%m%d%H%m%s)
-    docker-compose up --detach
-    docker compose logs -f >${playground_dir}/playground-${logSuffix}.log 2>&1 &
-    echo "Check log details: ${playground_dir}/playground-${logSuffix}.log"
+    logSuffix=$(date +%Y%m%d%H%M%s)
+    ${dockerComposeCommand} -p ${playgroundRuntimeName} up --detach
+    ${dockerComposeCommand} -p ${playgroundRuntimeName} logs -f >${playground_dir}/playground-${logSuffix}.log 2>&1 &
+    echo "[INFO] Check log details: ${playground_dir}/playground-${logSuffix}.log"
     ;;
   esac
 }
 
 status() {
-  case "$runtime" in
+  checkCurrentRuntime
+  if [ -z "${runtime}" ]; then
+    echo "[ERROR] Playground is not running."
+    exit 1
+  fi
+
+  case "${runtime}" in
   k8s)
     kubectl -n gravitino-playground get pods -o wide
```

Comment:
It's better to replace hard code namespace with ${playgroundRuntimeName} like the helm install command in start method

Reply:
fixed